src/ECER.Resources.Documents/Applications/ApplicationChildren/ProfessionalDevelopments.cs

```diff
@@ -149,6 +149,27 @@ private async Task AddFilesForProfessionalDevelopment(ecer_ProfessionalDevelopme
     }
   }
 
+  public async Task<string> AddProfessionalDevelopmentForSubmittedApplication(ProfessionalDevelopment newProfessionalDevelopment, string applicationId, string userId, CancellationToken cancellationToken)
+  {
+    await Task.CompletedTask;
+
+    var ecerApplication = context.ecer_ApplicationSet.FirstOrDefault(d => d.ecer_ApplicationId == Guid.Parse(applicationId) && d.StatusCode == ecer_Application_StatusCode.Submitted && d.ecer_Applicantid.Id == Guid.Parse(userId));
+
+    if (ecerApplication == null) throw new InvalidOperationException($"Application '{applicationId}' not found");
+
+    var ecerProfessionalDevelopment = mapper.Map<ecer_ProfessionalDevelopment>(newProfessionalDevelopment)!;
+    var newId = Guid.NewGuid();
+    ecerProfessionalDevelopment.ecer_ProfessionalDevelopmentId = newId;
```

Comment:
we could merge these two lines

Reply:
Fixed in https://github.com/bcgov/ECC-ECER/pull/521